stdlib/websub/src/main/ballerina/websub/commons.bal

```diff
@@ -440,63 +447,59 @@ public type SubscriptionChangeResponse record {
+# + mode - If remote publishing is allowed, the mode to use, `direct` (default) - fat ping with
+#                          the notification payload specified or `fetch` - the hub fetches the topic URL
+#                          specified in the "publish" request to identify the payload
+public type RemotePublishConfig record {
+    boolean enabled = false;
+    RemotePublishMode mode = PUBLISH_MODE_DIRECT;
+};
+
+# Starts up the Ballerina Hub.
+#
+# + hubServiceListener - The `http:Listener` to which the hub service is attached
+# + hubConfiguration - The hub specific configurations
 # + return - `WebSubHub` The WebSubHub object representing the newly started up hub, or `HubStartedUpError` indicating
 #            that the hub is already started, and including the WebSubHub object representing the
 #            already started up hub
-public function startHub(string? host = (), int port, int? leaseSeconds = (), string? signatureMethod = (),
-                         boolean? remotePublishingEnabled = (), RemotePublishMode? remotePublishMode = (),
-                         boolean? topicRegistrationRequired = (), string? publicUrl = (),
-                         boolean? sslEnabled = (), http:ServiceSecureSocket? serviceSecureSocket = (),
-                         http:SecureSocket? clientSecureSocket = ()) returns WebSubHub|HubStartedUpError {
-    hubHost = config:getAsString("b7a.websub.hub.host", default = host ?: DEFAULT_HOST);
-    hubPort = config:getAsInt("b7a.websub.hub.port", default = port);
+public function startHub(http:Listener hubServiceListener, HubConfiguration? hubConfiguration = ())
+                                                                    returns WebSubHub|HubStartedUpError {
     hubLeaseSeconds = config:getAsInt("b7a.websub.hub.leasetime",
-                                      default = leaseSeconds ?: DEFAULT_LEASE_SECONDS_VALUE);
-    hubSignatureMethod = config:getAsString("b7a.websub.hub.signaturemethod",
-                                   default = signatureMethod ?: DEFAULT_SIGNATURE_METHOD);
-    hubRemotePublishingEnabled = config:getAsBoolean("b7a.websub.hub.remotepublish",
-                                     default = remotePublishingEnabled ?: false);
-
-    string remotePublishModeAsConfig =  config:getAsString("b7a.websub.hub.remotepublish.mode");
-    if (remotePublishModeAsConfig == "") {
-        hubRemotePublishMode = remotePublishMode ?: PUBLISH_MODE_DIRECT;
-    } else {
-        if (remotePublishModeAsConfig.equalsIgnoreCase(REMOTE_PUBLISHING_MODE_FETCH)) {
-            hubRemotePublishMode = PUBLISH_MODE_FETCH;
-        } else if (!remotePublishModeAsConfig.equalsIgnoreCase(REMOTE_PUBLISHING_MODE_DIRECT)) {
-            log:printWarn("unknown publish mode: [" + remotePublishModeAsConfig + "], defaulting to direct mode");
-        }
-    }
-
+                                      default = hubConfiguration["leaseSeconds"] ?: DEFAULT_LEASE_SECONDS_VALUE);
```

Comment:
leaseSeconds will never be nil now right? 

Can't we say 
```ballerina
hubLeaseSeconds = config:getAsInt("b7a.websub.hub.leasetime", default = hubConfiguration.leaseSeconds);
```

Reply:
same for hubConfiguration["signatureMethod"] and hubConfiguration["topicRegistrationRequired"]

Reply:
Need to have nil lifiting there as hubConfiguration is nilable